spyder/plugins/editor/widgets/codeeditor/multicursor_mixin.py

```diff
@@ -0,0 +1,521 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+Mixin to manage editing with multiple cursors.
+"""
+
+# Standard library imports
+import functools
+import itertools
+
+# Third party imports
+from qtpy.QtCore import Qt, QTimer, Slot
+from qtpy.QtGui import (
+    QFontMetrics, QPaintEvent, QPainter, QTextCursor, QKeyEvent
+)
+from qtpy.QtWidgets import QApplication
+
+# Local imports
+from spyder.plugins.editor.api.decoration import TextDecoration
+
+
+class MultiCursorMixin:
+    """Mixin to manage editing with multiple cursors."""
+
+    def init_multi_cursor(self):
+        """Initialize attrs and callbacks for multi-cursor functionality"""
+        # actual default comes from setup_editor default args
+        self.multi_cursor_enabled = False
+        self.cursor_width = self.get_conf('cursor/width', section='main')
+        self.overwrite_mode = self.overwriteMode()
+
+        # Track overwrite manually when for painting reasons with multi-cursor
+        self.setOverwriteMode(False)
+        self.setCursorWidth(0)  # draw our own cursor
+        self.extra_cursors = []
+        self.cursor_blink_state = False
+        self.cursor_blink_timer = QTimer(self)
+        self.cursor_blink_timer.setInterval(
+            QApplication.cursorFlashTime() // 2
+        )
+        self.cursor_blink_timer.timeout.connect(
+            self._on_cursor_blinktimer_timeout
+        )
+        self.focus_in.connect(self.start_cursor_blink)
+        self.focus_changed.connect(self.stop_cursor_blink)
+        self.painted.connect(self.paint_cursors)
+        self.multi_cursor_ignore_history = False
+        self._drag_cursor = None
+
+    def __selection_colors(self, cache=[]):
+        """
+        Delayed retrival of highlighted text style colors. This is needed as
+        the palette hasn't been correctly set at the time init_multi_cursor is
+        called.
+        """
+        if not cache:
+            # Text Color
+            cache.append(self.palette().highlightedText().color())
+            # Background Color
+            cache.append(self.palette().highlight().color())
```

Comment:
I don't understand why you can't use `QColor(SpyderPalette.COLOR_TEXT_1)` and `QColor(SpyderPalette.COLOR_ACCENT_1)` here. Those are class constants, so they are available by simply importing `SpyderPalette` and then accessing them.

Reply:
I had run across that solution first, but I thought given unknown future changes to the theming system, pulling the colors from the widget directly might be better. If `SpyderPalette` is the way to go, then I'll go with that.